lib/neo4j/shared/enum.rb

```diff
@@ -0,0 +1,120 @@
+require 'active_support/core_ext/hash/except'
+require 'active_support/core_ext/hash/slice'
+
+module Neo4j::Shared
+  module Enum
+    extend ActiveSupport::Concern
+
+    class ConflictingEnumMethodError < Neo4j::Error; end
+
+    module ClassMethods
+      attr_reader :neo4j_enum_data
+
+      def enum(parameters = {})
+        options, parameters = *split_options_and_parameters(parameters)
```

Comment:
Sorry for not commenting earlier, didn't see I was tagged here. I don't really have a strong opinion either way. Is there any chance that breaking with convention would harm compatibility with gems that rely on ActiveRecord's implementation?